Demo/Common/IntegrationExamples/Components/ApplePayComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,193 @@
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+    private func presentAlert(with error: Error, retryHandler: (() -> Void)? = nil) {
+        presenter?.presentAlert(with: error, retryHandler: retryHandler)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PaymentComponentDelegate {
+
+    internal func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent) {
+        let request = PaymentsRequest(data: data)
+        apiClient.perform(request) { [weak self] result in
+            self?.paymentResponseHandler(result: result)
+        }
+    }
+
+    internal func didFail(with error: Error, from component: PaymentComponent) {
+        finish(with: error)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PresentationDelegate {
+
+    internal func present(component: PresentableComponent) {
+        let componentViewController = component.viewController
+        presenter?.present(viewController: componentViewController, completion: nil)
+    }
+}
+
+extension ApplePayComponentAdvancedFlowExample: ApplePayComponentDelegate {
+
+    func didUpdate(contact: PKContact,
+                   for payment: ApplePayPayment,
+                   completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void) {
+        var items = payment.summaryItems
+        print(items.reduce("> ") { $0 + "| \($1.label): \($1.amount.floatValue.rounded()) " })
+        if let last = items.last {
+            items = items.dropLast()
+            let cityLabel = contact.postalAddress?.city ?? "Somewhere"
+            items.append(.init(label: "Shipping \(cityLabel)",
+                               amount: NSDecimalNumber(value: 5.0)))
+            items.append(.init(label: last.label, amount: NSDecimalNumber(value: last.amount.floatValue + 5.0)))
+        }
+        completion(.init(paymentSummaryItems: items))
+    }
+
+    func didUpdate(shippingMethod: PKShippingMethod,
+                   for payment: ApplePayPayment,
+                   completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
+        var items = payment.summaryItems
+        print(items.reduce("> ") { $0 + "| \($1.label): \($1.amount.floatValue.rounded()) " })
+        if let last = items.last {
+            items = items.dropLast()
+            items.append(shippingMethod)
+            items.append(.init(label: last.label,
+                               amount: NSDecimalNumber(value: last.amount.floatValue + shippingMethod.amount.floatValue)))
+        }
+        completion(.init(paymentSummaryItems: items))
+    }
+
+    @available(iOS 15.0, *)
+    func didUpdate(couponCode: String,
+                   for payment: ApplePayPayment,
+                   completion: @escaping (PKPaymentRequestCouponCodeUpdate) -> Void) {
+        var items = payment.summaryItems
+        print(items.reduce("> ") { $0 + "| \($1.label): \($1.amount.floatValue.rounded()) " })
```

Comment:
Is this debug print statement needed still?

Reply:
930bacee2331858be5473647d4406808c9d72717